### PR TITLE
Hotfix for spot TPU pod recovery

### PR DESCRIPTION
**Pull request by infwinston**
A hotfix to temporarily close https://github.com/skypilot-org/skypilot/issues/1468. We may need to modify the controller logic to make it more future proof. 

The issue is: When preemption happens, our spot controller won't cleanup preempted VMs before running another `sky launch`. This causes two VMs to co-exist and both the preempted one and the recovered one will be returned.

### CODE REVIEW

File: sky/backends/cloud_vm_ray_backend.py
Comment (concretevitamin) [inline sky/backends/cloud_vm_ray_backend.py:2627]: What happens to preempted TPUs? Is it true that they require no further cleanup actions -- i.e., `gcloud compute tpus tpu-vm delete` -- from us (e.g., disks)? Worth a comment!

**Review by concretevitamin — COMMENTED** [review]
What's the issue? Probably @Michaelvll knows more context.

File: sky/backends/backend_utils.py
Comment (Michaelvll) [inline sky/backends/backend_utils.py:1226]: I think this makes sense. 

A minor thing that may not be good to fix in this PR is that in L1150, we directly return the IPs, without checking whether the number of nodes in the cluster matches the expected nodes as we done in L1209. Can we raise the `exceptions.FetchIPError(.FetchIPError.Reason.HEAD)` instead of handling it separately below?
https://github.com/skypilot-org/skypilot/blob/2abcde326caff998be60b788276de25afaacb500/sky/backends/backend_utils.py#L1620-L1622

Also, `get_node_ips` also checks the ray cluster is correctly running on the cluster. Is it possible for TPU VM having multiple nodes? If yes, we may want to check the healthiness of the ray cluster as well.

**Review by Michaelvll — COMMENTED** [review]
Thanks for the quick fix @infwinston!

File: sky/backends/cloud_vm_ray_backend.py
Comment (infwinston) [reply sky/backends/cloud_vm_ray_backend.py:2629]: good point I just updated some comments.

According to [reference](https://cloud.google.com/tpu/docs/setup-persistent-disk#set-up-pd-with-tpu), TPU VM's disk is not persistent unless manually specifying it.
So a preempted TPU VM will cost zero and should be safe to ignore for cost perspective. (IIRC, GCP will clean them up after a while)

However, I agree we should try to cleanup the preempted VMs if possible, which requires changing the logic in spot controller to be tearing down VM first and then `sky launch` again.

For now I just wanted to ship to wilson asap. I'll create another PR to fix this.

File: sky/backends/backend_utils.py
Comment (infwinston) [reply sky/backends/backend_utils.py:1228]: Ah yes, we should make the behavior consistent and a part of the issue is related to https://github.com/skypilot-org/skypilot/pull/1185.
I'll make sure that PR fixes it and merge it asap.

**Comment by infwinston** [comment]
> What's the issue? Probably @Michaelvll knows more context.

Ah sorry the issue is: When preemption happens, our spot controller won't cleanup preempted VMs before running another sky launch. This causes two VMs to co-exist and both the preempted one and the recovered one will be returned.

**Review by Michaelvll — APPROVED** [review]
LGTM. We may want to test the autostop and `sky status -r` for the TPU VM before merging it. :)

**Comment by infwinston** [comment]
Just tested `sky status -r` works. Offline discussion with @Michaelvll: we should block `sky autostop` spot VMs given that `sky stop` a spot VM is not supported.

**Comment by infwinston** [comment]
Merging this for now to unblock our user. Also will work on a long-term fix which requires changing some node provider logic.

**Pull request merged**
